src/main/java/at/hannibal2/skyhanni/features/misc/HideFarEntities.kt

```diff
@@ -0,0 +1,40 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.CheckRenderEntityEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.utils.CollectionUtils.sorted
+import at.hannibal2.skyhanni.utils.EntityUtils
+import at.hannibal2.skyhanni.utils.LocationUtils.distanceToPlayer
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class HideFarEntities {
+    private val config get() = SkyHanniMod.feature.dev.hideFarEntities
+
+    private var ignored = emptySet<Int>()
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+
+        val maxAmount = config.maxAmount.coerceAtLeast(1)
+        val minDistance = config.minDistance.coerceAtLeast(3)
+
+        ignored = EntityUtils.getAllEntities()
+            .map { it to it.distanceToPlayer() }
+            .toMap()
+            .filter { it.value > minDistance }
+            .sorted().keys.drop(maxAmount)
+            .map { it.entityId }.toSet()
```

Comment:
I have two versions to improve this. First one is 30% faster and keeps the same readability. Second one is 40% faster and makes a bit of a mess (readability wise).
You choose.
```suggestion
        ignored = EntityUtils.getAllEntities()
            .map { it.entityId to it.distanceToPlayer() }
            .filter { it.second > minDistance }
            .sortedBy { it.second }.drop(maxAmount)
            .map { it.first }.toSet()
```
```suggestion
        ignored = EntityUtils.getAllEntities()
            .fold(mutableListOf<Pair<Int, Double>>())
            { list, it ->
                val distance = it.distanceToPlayer()
                if (distance > minDistance) {
                    list.add(it.entityId to distance)
                }
                list
            }.sortedBy { it.second }.drop(maxAmount)
            .map { it.first }.toSet()
```